pycontrails/models/dry_advection.py

```diff
@@ -0,0 +1,430 @@
+"""Simulate dry advection."""
+
+from __future__ import annotations
+
+import dataclasses
+from typing import Any, NoReturn, overload
+
+import numpy as np
+
+from pycontrails.core import models
+from pycontrails.core.flight import Flight
+from pycontrails.core.met import MetDataset
+from pycontrails.core.met_var import AirTemperature, EastwardWind, NorthwardWind, VerticalVelocity
+from pycontrails.core.vector import GeoVectorDataset
+from pycontrails.models.cocip import contrail_properties, wind_shear
+from pycontrails.physics import geo, thermo
+
+
+@dataclasses.dataclass
+class DryAdvectionParams(models.ModelParams):
+    """Parameters for the :class:`DryAdvection` model."""
+
+    #: Apply Euler's method with a fixed step size of ``dt_integration``. Advected waypoints
+    #: are interpolated against met data once each ``dt_integration``.
+    dt_integration: np.timedelta64 = np.timedelta64(30, "m")
+
+    #: Max age of contrail evolution.
+    max_age: np.timedelta64 = np.timedelta64(20, "h")
+
+    #: Difference in altitude between top and bottom layer for stratification calculations,
+    #: [:math:`m`]. Used to approximate derivative of "lagrangian_tendency_of_air_pressure" layer.
```

Comment:
We should add more here since "lagrangian tendency..." is opaque.  What about

> Used to approximate derivative of ``lagrangian_tendency_of_air_pressure`` (upward component of air velocity).

Reply:
c69d26b1e2af7af0abaf07a9414ea71d8550007d